homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,221 @@
+"""Config flow for generic (IP Camera)."""
+from __future__ import annotations
+
+from functools import partial
+import imghdr
+import logging
+from typing import Any
+
+import av
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.stream.const import SOURCE_TIMEOUT
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+    HTTP_BASIC_AUTHENTICATION,
+    HTTP_DIGEST_AUTHENTICATION,
+)
+
+from .camera import DEFAULT_CONTENT_TYPE, GenericCamera
+
+# pylint: disable=unused-import
+from .const import (
+    CONF_CONTENT_TYPE,
+    CONF_FRAMERATE,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+    CONF_RTSP_TRANSPORT,
+    CONF_STILL_IMAGE_URL,
+    CONF_STREAM_SOURCE,
+    DEFAULT_NAME,
+    DOMAIN,
+    FFMPEG_OPTION_MAP,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DATA = {
+    CONF_NAME: DEFAULT_NAME,
+    CONF_AUTHENTICATION: HTTP_BASIC_AUTHENTICATION,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE: False,
+    CONF_CONTENT_TYPE: DEFAULT_CONTENT_TYPE,
+    CONF_FRAMERATE: 2,
+    CONF_VERIFY_SSL: True,
+}
+
+
+class GenericIPCamConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for generic IP camera."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    async def _test_connection(self, info) -> tuple[bool, str]:
+        """Verify that the camera data is valid before we add it."""
+        unique_id = self.flow_id
+        if unique_id in self._async_current_ids():
+            return False, "unique_id_already_in_use"
+        await self.async_set_unique_id(unique_id)
+        cam = GenericCamera(self.hass, info, unique_id)
+        if info.get(CONF_STILL_IMAGE_URL) not in [None, ""]:
+            # First try getting a still image
+            image = await cam.async_camera_image()
+            if image is None:
+                return False, "unable_still_load"
+            fmt = imghdr.what(None, h=image)
+            _LOGGER.debug(
+                "Still image at '%s' detected format: %s",
+                info[CONF_STILL_IMAGE_URL],
+                fmt,
+            )
+            image_type = info.get(CONF_CONTENT_TYPE)
+            if image_type is not None and image_type not in ["image/jpeg", "image/png"]:
```

Comment:
```suggestion
            if image_type is not None and image_type not in SUPPORTED_IMAGE_TYPES:
```

Add `SUPPORTED_IMAGE_TYPES = {"image/jpeg", "image/png"}` above

Reply:
Good idea, done.